add support for linkvertise pastes

davidkra230 opened the pull request:
<!-- A link to all issues that this pull request will address, 
Link all issues with the number, like #123. Put "None" if you are making a new bypass-->
Fixes (Links to issues fixed by this PR): 
None
<!-- A brief description of what you did. Write the sites you bypassed and what changes/ additions to the source code.
Don't worry about this being too long, we care more about the code than your writing skills 😉-->
Description:
bypasses linkvertise's pasting feature. basically i just added a check for pastes and if it matches it will get the info almost exactly like the url bypass except it copies the paste instead of redirecting.
<!-- Add test links for all sites in the pull request. Make sure to link one test link per site
Make sure to hyperlink test links to avoid making the PR look messy
To make a hyperlink, the format is [domain name](direct link)-->
Test links:
[LinkvertisePaste](https://linkvertise.com/114822/test?o=sharing)
Checklist:
<!--Add an x to mark as done-->
- [x] I made sure there are no unnecessary changes in the code*
- [x] Tested on Chromium- Browser OS
- [ ] Tested on Firefox

<!--\* indicates required -->

lem6ns commented on src/js/injection_script.js:
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
+		    if (data.currentTarget.responseText.includes("tokens")) {
+			const response = JSON.parse(data.currentTarget.responseText)
+			if (!response.data.valid)
+			    return insertInfoBox(
+				"Please solve the captcha, afterwards we can immediately redirect you"
+			    )
```
why was formatting changed here?

davidkra230 replied:
prettier was used

lem6ns commented on src/js/injection_script.js:
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
+
+			const target_token = response.data.tokens["TARGET"]
+			const ut = localStorage.getItem("X-LINKVERTISE-UT")
+			const linkvertise_link = location.pathname.replace(
+			    /\/[0-9]$/,
+			    ""
+			)
```
why was formatting changed here?

davidkra230 replied:
prettier was used

lem6ns commented on src/js/injection_script.js:
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
+			    .then((r) => r.json())
+			    .then((json) => {
+				if (json?.data.link.target_type !== "URL") {
+				    insertInfoBox("Paste detected, bypassing..")
```
this line isn't needed, and what if `.target_type` was also not a paste?

lem6ns commented on src/js/injection_script.js:
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
+				if (json?.data.link.target_type !== "URL") {
+				    insertInfoBox("Paste detected, bypassing..")
+				    // what is referred to as the pathmap is a table of functions to bypass for each type of content.
+				    let pathmap = {
```
change to `const` instead of `let`

lem6ns commented on src/js/injection_script.js:
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
+					"PASTE": function () {
+					    if (json?.data.link.id) {
+						const json_body = {
+						    serial: btoa(
+							JSON.stringify({
+							    timestamp:
+								new Date().getTime(),
+							    random: "6548307",
+							    link_id: json.data.link.id,
+							})
```
consider making an inner function for creating a serial

lem6ns commented on src/js/injection_script.js:
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
+							body: JSON.stringify(json_body),
+							headers: {
+							    Accept: "application/json",
+							    "Content-Type":
+								"application/json",
```
awkward formatting?

davidkra230 replied:
prettier was used

lem6ns commented on src/js/injection_script.js:
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
+						    .then((r) => r.json())
+						    .then((json) => {
+							if (json?.data.paste) {
+							    insertInfoBox("Bypassed, attempting to copy to clipboard.")
+							    navigator.clipboard.writeText(json?.data.paste).then(function() {
+							      insertInfoBox("Copied paste to clipboard!\nYou may now close this tab.")
+							    }, (err) => {
+							      window.prompt("Copying to clipboard failed!\nThe paste is displayed below. Ctrl+C to copy it.", json?.data.paste)
```
consider creating a blob text file, then creating an objectURL to redirect/open a new tab to.

lem6ns commented on src/js/injection_script.js:
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
+				}
+				if (json?.data.link.id) {
+				    const json_body = {
+					serial: btoa(
+					    JSON.stringify({
+						timestamp: new Date().getTime(),
+						random: "6548307",
+						link_id: json.data.link.id,
+					    })
```
consider making an inner function for creating a serial

lem6ns reviewed (changes requested):
please fix formatting and see review comments.

davidkra230 commented:
I have fixed these but, I had deleted the original repository in which I had made the changes and therefore, I need to open a new pull request.

I will close this pull request for now, and I will open a new one with the requested changes.

lem6ns commented:
alright
